gap/FinSetsForCAP.gi

```diff
@@ -1034,3 +1045,36 @@ InstallMethod( Display,
   function ( phi )
     Print( [ AsList( Source( phi ) ), AsList( phi ), AsList( Range( phi ) ) ], "\n" );
 end );
+
+##
+BindGlobal( "FinSets", CategoryOfFinSets( ) );
+
+##
+InstallMethod( FinSetNC,
+        "for a list",
+        [ IsList ],
+        
+  function ( L )
+    local set, i;
+    
+    set := rec( );
+    
+    ObjectifyObjectForCAPWithAttributes( set, FinSets,
```

Comment:
If you allow to create multiple instances of the category of finite sets via the new constructor `CategoryOfFinSets`, the global variable `FinSets` must not be hardcoded here. We need a new version of `FinSetNC` with the category as the first argument. This version has to be used throughout the implementation above. Then the code of `FinSetNC` can probably also be moved to the top again.